lib/controllers/recipe_bloc.dart

```diff
@@ -0,0 +1,18 @@
+import 'package:bloc/bloc.dart';
+import 'recipe_event.dart';
+import 'recipe_state.dart';
+import '../services/recipe_service.dart';
+
+class RecipeBloc extends Bloc<RecipeEvent, RecipeState> {
+  RecipeBloc() : super(RecipeInitial()) {
+    on<FetchRecipes>((event, emit) async {
+      emit(RecipeLoading());
+      try {
+        final recipes = await RecipeService.fetchRecipes();
+        emit(RecipeLoaded(recipes));
+      } catch (e) {
+        emit(RecipeError(e.toString()));
+      }
+    });
+  }
+}
```

Comment:
All dependencies (`RecipeService`) should be inject via constructor and avoid creating a static method since it's unable to mock when we're writing test.
```suggestion
class RecipeBloc extends Bloc<RecipeEvent, RecipeState> {
  RecipeBloc({
    RecipeService? recipeService,
  })  : _recipeService = recipeService ?? RecipeService(),
        super(RecipeInitial()) {
    on<FetchRecipes>((event, emit) async {
      emit(RecipeLoading());
      try {
        final recipes = await _recipeService.fetchRecipes();
        emit(RecipeLoaded(recipes));
      } catch (e) {
        emit(RecipeError(e.toString()));
      }
    });
  }

  final RecipeService _recipeService;
}
```